crates/file_store/src/store.rs

```diff
@@ -372,17 +436,26 @@ mod test {
                     }),
                     "should recover all changesets that are written in full",
                 );
-                db.db_file.write_all(&last_changeset_bytes).unwrap();
+                // Remove file and start again
+                fs::remove_file(&file_path).expect("should remove file");
+                let mut store =
+                    Store::<TestChangeSet>::create(&TEST_MAGIC_BYTES, &file_path).unwrap();
+                for changeset in &changesets {
+                    store.append(changeset).unwrap();
+                }
+                // this is the incomplete write
```

Comment:
This is the complete write, no?

It's a little unfortunate that we can only recover from a corrupt file by removing it and starting again. To better simulate a recovery you could use the agg. changeset given by the load error to write back to the new file. But I think another solution is to have `load` return a `Store` instance with the partially aggregated changeset even in case of a bincode error. Then we can just append the `last_changeset` to the existing store. What do you think?

Reply:
Gotcha. It seems like either way changesets 3, 4, and 5 will be unrecoverable unless they're somehow predetermined ahead of time?

> I don't think so, the following line just writes the changeset bytes up to the short_write_len limit, which will produce a partial write. There's something I'm not seeing?

Hm, I was thinking that the short write was above on L412

Reply:
> Gotcha. It seems like either way changesets 3, 4, and 5 will be unrecoverable unless they're somehow predetermined ahead of time?

Yes, they will be unrecoverable, but I don't get what you mean by "predetermined ahead of time".

> > I don't think so, the following line just writes the changeset bytes up to the short_write_len limit, which will produce a partial write. There's something I'm not seeing?
> 
> Hm, I was thinking that the short write was above on L412

Indeed, there was something I was not seeing. You're right, it is the complete write. I was matching the comment with the wrong line.